editor/plugins/script_text_editor.cpp

```diff
@@ -1598,8 +1598,27 @@ static Node *_find_script_node(Node *p_edited_scene, Node *p_current_node, const
 	return nullptr;
 }
 
+
+String ScriptTextEditor::_escape_drop_data(const String &str) {
+	// This function escapes resource paths, node paths, or object properties
+	// according to String delimiter preferences.
+	
+	String escaped = str.c_escape();
+
+	// Resource paths, node paths and object properties cannot contain double-quotes
+	// so we do not bother reverting back the .replace from c_escape.
+	
+	if (!EDITOR_GET("text_editor/completion/use_single_quotes")) {
+		// Only revert single-quotes if we're using double-quotes as string delimiters.
+		escaped = escaped.replace("\\'", "\'"); 
+	}
+
+	return escaped;
+}
+
 void ScriptTextEditor::drop_data_fw(const Point2 &p_point, const Variant &p_data, Control *p_from) {
-	const String quote_style = EDITOR_GET("text_editor/completion/use_single_quotes") ? "'" : "\"";
+	const bool using_single_quotes = EDITOR_GET("text_editor/completion/use_single_quotes");
+	const String quote_style = using_single_quotes ? "'" : "\"";
```

Comment:
Can we remove the need for this and put `.quote` inside  `_escape_drop_data`?

Reply:
Yes, excellent suggestion! Changed _escape_drop_data to _quote_drop_data which already escapes and quotes as needed!